scripts.js

```diff
@@ -266,25 +258,53 @@ function handleDropdownRegion() {
     fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
   }
   
-  // Get currently selected region if in the same region page based on localeHome value
-  const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
-  if (!currentRegion.length) {
-    // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use language
-    sessionStorage.getItem('blog-selected-language') || window.blog.language;
-  } else {
-      // uk and apac, just save the info in the session storage
-      sessionStorage.setItem('blog-selected-language', currentRegion[0].lang);
+  // get actual selected Region
+  const selectedRegion = () => {
+    const regionPage = regionsNameList.find(r => r.localeHome === location.pathname);
+    // check if localeName matches the blog.language example 'en' in order to show the localeName in the buttom
+    const nonRegionPage = regionsNameList.find(r => r.lang === window.blog.language);
+    let selectedLocale;
+    let selectedLocaleName;
+    if (!regionPage) {
+      // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use blog.language
+      selectedLocale = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+      // in order to show the localeName either from SessionStorage locale or blog.language in our region button
+      if (selectedLocale !== window.blog.language) {
+        // if the selected locale do not match the blog.language we get the LocaleName name based on the sessionStorage saved Language Value
+        const storedLanguage = regionsNameList.find(r => r.lang === selectedLocale);
+        selectedLocaleName = storedLanguage.localeName;
+      } else {
+        // else we get the localeName base on the blog.language and after have find the same language in our RegionListName
+        selectedLocaleName = nonRegionPage.localeName;
+      }
+    } else {
+      selectedLocale = regionPage.lang;
+      selectedLocaleName = regionPage.localeName;
+      // sessionStorage will be used only if current region lang is not same as blog.lang en_apac en_uk 
+      // where blog.lang will still being en
+      if (regionPage.lang !== window.blog.language) {
+        sessionStorage.setItem('blog-selected-language', regionPage.lang);
+      }
+    } return {selectedLocale, selectedLocaleName};
+  }
+
+  
+  const dropdownRegionList = document.querySelector('.region-dropdown-list');
+  const {selectedLocale, selectedLocaleName} = selectedRegion(); 
+
+  // Change locale value from Feds Region Picker Button adding a Region Locale Name 
+  const FEDSregionPickerText = document.querySelector('.feds-regionPicker-text');
+  if (FEDSregionPickerText && selectedLocaleName !== undefined) {
+      FEDSregionPickerText.innerText = selectedLocaleName;
   }
```

Comment:
I added it because if the case come that the user access the blog from a direct link like http://localhost:3000/de/topics/creativity.html# that actually is not on the regionListName now.
the button will show then the normal Feds Change region text in their language.
so we get:
![Screen Shot 2021-02-05 at 12 22 56 PM](https://user-images.githubusercontent.com/62952234/107027788-ea6d4e80-67ac-11eb-8bc5-ba8ae3835655.png)

Reply:
The accessing directly a `/de` link is not a supported use case yet. How would that help ? The "name" in that case is anyway unknown since `de` is not in the link. For now and until we add German as a supported "region", following back to the language is completely fine.

Reply:
It will help to show the button as Region ändern instead of undefined.

wdyt @kptdobe ?

Reply:
Well, if you go to `/de`, I am sure `selectedLocaleName` is `undefined` thus this code is useless in that case. And in the case of a known locale, the `selected` value of the combo is doing the selected. Thus I think you definitively do not need that code.

Reply:
removed @kptdobe

Reply:
ok, thanks. But please test your code, I do not see how the `getSelectedRegion` function can work with the `/de` use case.
If it works, I then approve the PR.

Reply:
wait, let me understand. You told "The accessing directly a /de link is not a supported use case yet." 
but saying that following back to the language we will have lang DE, but no regionName as we are not adding it right now.
so if /de is a use case but the region name is not being added to the code now, we do need the extra `if` 
or how do you see it. I got bit confused myself now @kptdobe

Reply:
Sorry if this is not clear.

You mentioned the `/de` case and it is great to think about it. By `/de` is not a supported region, I ment "nobody is supposed to go to the DE site and expect to see the "German" region selected (the "Region andern" you proposed is perfect) but it does not mean there should be a Javascript error when loading the page which I see on https://region-picker-cleanup--theblog--elaineskpt.hlx.page/de/index.html.